### PR TITLE
Mock payload plugin

### LINKED ISSUE #3390: Define payloads plugin interface
## Description

In order to create payloads plugins, we must first define a plugin interface for payloads.

## Tasks
- [x] Define what should be passed to `Plugin.__init__()` for payloads (0d) @shreyamalviya 
- [x] Define what should be passed to `Plugin.run()` for payloads plugins and modify the puppet as necessary (0d) @shreyamalviya 
- [x] Define what should be returned from `Plugin.run()` (if anything) for payloads and modify the puppet as necessary (0d) @shreyamalviya 
- [x] Modify `PluginLoader._load_plugin_from_island()` to load payload plugins (0d) @shreyamalviya 
    - [x] Modify puppet to verify OS-compatibility before running payloads
- [x] Make the UI work for payload plugins (0d) @shreyamalviya 
- [x] Build a mock/test payload plugin and test the system end-to-end (0d) @shreyamalviya 
- [ ] Consolidate `build.sh` scripts in `data_for_tests/` (0.25d) @shreyamalviya

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes #3390

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Do all unit tests pass?
* [ ] ~Do all end-to-end tests pass?~
* [x] Any other testing performed?
    > Tested Island
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/infection_monkey/i_puppet/payload_result_data.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class PayloadResultData:
```
Comment (mssalvatore) [inline monkey/infection_monkey/i_puppet/payload_result_data.py:5]: What about 
```suggestion
class PayloadResult:
```

File: monkey/tests/data_for_tests/payload_plugin/build.sh
Comment (mssalvatore) [inline monkey/tests/data_for_tests/payload_plugin/build.sh:1]: How does this differ from the other `build.sh` scripts in `data_for_tests`? Can we consolidate them?

**Review by mssalvatore — COMMENTED** [review]
What testing was performed?

**Comment by shreyamalviya** [comment]
> What testing was performed?

Verified that the Island loads the plugin from the data dir and shows it in the UI as expected

File: monkey/infection_monkey/i_puppet/payload_result_data.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/infection_monkey/i_puppet/payload_result_data.py:5]: I don't mind, but we have `ExploiterResultData`, so `PayloadResultData` would be more consistent.

Comment (mssalvatore) [reply monkey/infection_monkey/i_puppet/payload_result_data.py:5]: Ok. I think we should change them both. Followup PR? It requires rebuilding plugins (probably).

**Comment by mssalvatore** [comment]
> > What testing was performed?
> 
> Verified that the Island loads the plugin from the data dir and shows it in the UI as expected

Can we test that the agent runs the payload? Or is not all the infrastructure in place on the agent side?

File: monkey/infection_monkey/i_puppet/payload_result_data.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/infection_monkey/i_puppet/payload_result_data.py]: Changed `PayloadResultData`, opened #3532 for `ExploiterResultData`

File: monkey/tests/data_for_tests/payload_plugin/build.sh
Comment (shreyamalviya) [reply monkey/tests/data_for_tests/payload_plugin/build.sh:1]: Added a task to #3390 about this

**Comment by shreyamalviya** [comment]
> Can we test that the agent runs the payload? Or is not all the infrastructure in place on the agent side?

Required a small change, pushed a commit. The mock plugin works as expected. 
image

**Pull request merged**